packages/server/src/synchronizer/index.ts

```diff
@@ -58,6 +59,28 @@ export async function synchronizeFiles({
 
   await clean(destPath)
 
+  const allRoutes = entries.filter((route) => route.includes('routes'))
+  const cleanRoutes = allRoutes.map((route) => route.split('routes/')[1])
+  const duplicateRoutes = cleanRoutes.filter((e, i, a) => a.indexOf(e) !== i)
+  const duplicateRoutesIndexes = duplicateRoutes.map((route) =>
+    cleanRoutes.flatMap((cleanRoute, i) => (cleanRoute === route ? i : [])),
+  )
+  const duplicateRoutesPaths = duplicateRoutesIndexes.map((indexes) =>
+    indexes.map((index) => allRoutes[index]),
+  )
+
+  if (duplicateRoutesPaths.length > 0) {
+    console.log(chalk.yellow('Warning: You have created conflicting routes: \n'))
+
+    duplicateRoutesPaths.forEach((route) => {
+      route.forEach((path) =>
+        console.log(`- ${path.split('routes/')[0]}routes/${chalk.bold.redBright(path.split('routes/')[1])}`),
+      )
+    })
+
+    process.exit(0)
+  }
+
```

Comment:
This drafts this out and is really helpful. 

Couple of things I am thinking:
1. What do you think about putting this all into its own function?
1. As this is an invariant should this all happen before everything else?
1. Can/should this be generalised? precondition throws error which is caught and then displayed nicely. The answer could be no/later.

Reply:
Done 👍